lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -0,0 +1,715 @@
+    #[traced_test]
+    async fn decode_chunk_with_invalid_header() {
+        let mut src = BytesMut::new();
+        src.extend_from_slice(&GELF_MAGIC);
+        // Invalid chunk header with less than 10 bytes
+        let invalid_chunk = [0x12, 0x34];
+        src.extend_from_slice(&invalid_chunk);
+        let mut decoder = ChunkedGelfDecoder::default();
+        let frame = decoder.decode_eof(&mut src);
+
+        let error = frame.expect_err("Expected an error");
+        let downcasted_error = downcast_framing_error(&error);
+        matches!(
+            downcasted_error,
+            ChunkedGelfDecoderError::InvalidChunkHeader { .. }
+        );
+    }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn decode_chunk_with_invalid_total_chunks() {
+        let message_id = 1u64;
+        let sequence_number = 1u8;
+        let invalid_total_chunks = GELF_MAX_TOTAL_CHUNKS + 1;
+        let payload = "foo";
+        let mut chunk = create_chunk(message_id, sequence_number, invalid_total_chunks, payload);
+        let mut decoder = ChunkedGelfDecoder::default();
+
+        let frame = decoder.decode_eof(&mut chunk);
+        let error = frame.expect_err("Expected an error");
+        let downcasted_error = downcast_framing_error(&error);
+        matches!(
+            downcasted_error,
+            ChunkedGelfDecoderError::InvalidTotalChunks { .. }
+        );
+    }
+
+    #[tokio::test]
+    #[traced_test]
+    async fn decode_chunk_with_invalid_sequence_number() {
+        let message_id = 1u64;
+        let total_chunks = 2u8;
+        let invalid_sequence_number = total_chunks + 1;
+        let payload = "foo";
+        let mut chunk = create_chunk(message_id, invalid_sequence_number, total_chunks, payload);
+        let mut decoder = ChunkedGelfDecoder::default();
+
+        let frame = decoder.decode_eof(&mut chunk);
+        let error = frame.expect_err("Expected an error");
+        let downcasted_error = downcast_framing_error(&error);
+        matches!(
+            downcasted_error,
+            ChunkedGelfDecoderError::InvalidSequenceNumber { .. }
+        );
+    }
+
+    #[rstest]
+    #[tokio::test]
+    #[traced_test]
+    async fn decode_when_reached_pending_messages_limit(
+        two_chunks_message: ([BytesMut; 2], String),
+        three_chunks_message: ([BytesMut; 3], String),
+    ) {
+        let pending_messages_limit = 1;
+        let (mut two_chunks, _) = two_chunks_message;
+        let (mut three_chunks, _) = three_chunks_message;
+        let mut decoder = ChunkedGelfDecoder::new(DEFAULT_TIMEOUT_MILLIS, pending_messages_limit);
+
+        let frame = decoder.decode_eof(&mut two_chunks[0]).unwrap();
+        assert!(frame.is_none());
+        assert!(decoder.state.lock().unwrap().len() == 1);
+
+        let frame = decoder.decode_eof(&mut three_chunks[0]);
+        let error = frame.expect_err("Expected an error");
+        let downcasted_error = downcast_framing_error(&error);
+        matches!(
```

Comment:
I could have done just `matches!(frame, Err(_))`, but then I wouldn't be testing that it returns the concrete enum variant, I think it's better with this "hack"